app/src/main/java/software/amazon/freertos/demo/DeviceScanFragment.java

```diff
@@ -67,63 +69,67 @@ public BleDeviceHolder(LayoutInflater inflater, ViewGroup parent) {
-                    Log.i(TAG, "Connect switch isChecked: " + (isChecked ? "ON":"OFF"));
-                    if (isChecked) {
-                        if (aDevice == null) {
-                            AWSCredentialsProvider credentialsProvider = AWSMobileClient.getInstance();
-
-                            aDevice = mAmazonFreeRTOSManager.connectToDevice(mBleDevice.getBluetoothDevice(),
-                                    connectionStatusCallback, credentialsProvider, autoReconnect);
-                        }
-                    } else {
-                        if (userDisconnect || !autoReconnect) {
-                            if (aDevice != null) {
-                                mAmazonFreeRTOSManager.disconnectFromDevice(aDevice);
-                                aDevice = null;
-                            }
+        private final CompoundButton.OnCheckedChangeListener deviceConnectListener =
+                new CompoundButton.OnCheckedChangeListener() {
+                    @Override
+                    public void onCheckedChanged(CompoundButton v, boolean isChecked) {
+                        Log.i(TAG, "Connect switch isChecked: " + (isChecked ? "ON" : "OFF"));
+                        if (isChecked) {
+                            connectDevice();
                         } else {
-                            userDisconnect = true;
+                            disconnectDevice();
+                            resetUI();
                         }
-                        resetUI();
                     }
-                }
-            });
+        };
 
-            mMenuTextView.setOnClickListener(new View.OnClickListener(){
-                @Override
-                public void onClick(View view) {
-                    Log.i(TAG, "Click menu.");
-                    PopupMenu popup = new PopupMenu(getContext(), mMenuTextView);
-                    popup.inflate(R.menu.options_menu);
-                    popup.setOnMenuItemClickListener(new PopupMenu.OnMenuItemClickListener() {
-                        @Override
-                        public boolean onMenuItemClick(MenuItem item) {
-                            switch (item.getItemId()) {
-                                case R.id.wifi_provisioning_menu_id:
-                                    Intent intentToStartWifiProvision
-                                            = WifiProvisionActivity.newIntent(getActivity(), mBleDevice.getMacAddr());
-                                    startActivity(intentToStartWifiProvision);
-                                    return true;
-                                case R.id.mqtt_proxy_menu_id:
-                                    Toast.makeText(getContext(),"Already signed in",Toast.LENGTH_SHORT).show();
-                                    return true;
-                            }
-                            return false;
+        private void connectDevice() {
```

Comment:
Depending on response to my first comment above, this might need to instead return data to indicate pass/fail for establishing connection

Reply:
As discussed above a return value is not necessary here as the disconnected callback is triggered if it can't connect to the device.